pkg/cfn/manager/api.go

```diff
@@ -104,6 +108,30 @@ func (c *StackCollection) CreateStack(name string, stack builder.ResourceSet, pa
 	return nil
 }
 
+// UpdateStack will update a cloudformation stack. It uses changesets and if in debug it will log the changes.
+// This is used bu things like nodegroup scaling
+func (c *StackCollection) UpdateStack(stackName string, action string, description string, template []byte, parameters map[string]string) error {
+	i := &Stack{StackName: &stackName}
+	changesetName, err := c.doCreateChangesetRequest(i, action, description, template, parameters, true)
+	if err != nil {
+		return err
+	}
+	err = c.doWaitUntilChangeSetIsCreated(i, &changesetName)
+	if err != nil {
+		return err
+	}
+	changeset, err := c.describeStackChangeset(i, &changesetName)
+	if err != nil {
+		return err
+	}
+	logger.Debug("changes = %#v", changeset.Changes)
+	if err := c.doExecuteChangeset(stackName, changesetName); err != nil {
+		logger.Warning("error executing Cloudformation changeset %s in stack %s. Check the Cloudformation console for further details", changesetName, stackName)
```

Comment:
'existing event tracer' - is this in relation to the addition of `waitWithAcceptorsChangeSet` and instead could we re-use `waitWithAcceptors`? If thats the case then we could generalise these down into 1 function but i felt that this would then make the the `doWaitUntil*` functions more complicated. Happy to generlise this doen to 1 function if you think its best?

Reply:
I am thinking of code that was added in #202. Namely, can we do anything before suggesting users to go to CloudFormation console (as you know, it doesn't exactly tell you what the issue is at glance, so it's not the most helpful). I'll read through the code once again, perhaps it's already hits `c.troubleshootStackFailureCause` by this point...

Reply:
The following `doWaitUntilStackIsUpdated` will hit `c.troubleshootStackFailureCause` but `c.doExecuteChangeset` won't. This error handler should only be hit if we couldn't create the changeset (whether thats a validation error with the request or with communicating with the AWS endpoint).